src-java/base-topology/base-messaging/src/test/java/org/openkilda/messaging/info/switches/v2/GroupsValidationEntryV2Test.java

```diff
@@ -0,0 +1,207 @@
+/* Copyright 2022 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.messaging.info.switches.v2;
+
+import static org.junit.Assert.assertEquals;
+
+import org.openkilda.messaging.info.switches.v2.GroupInfoEntryV2.BucketEntry;
+
+import com.google.common.collect.Lists;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GroupsValidationEntryV2Test {
+
+    @Test
+    public void splitAndUniteEmptyGroupEntryTest() {
+        GroupsValidationEntryV2 entry = GroupsValidationEntryV2.builder()
+                .asExpected(true)
+                .missing(new ArrayList<>())
+                .misconfigured(new ArrayList<>())
+                .excess(new ArrayList<>())
+                .proper(new ArrayList<>())
+                .build();
+        List<GroupsValidationEntryV2> list = entry.split(4, 4);
+        assertEquals(1, list.size());
+        GroupsValidationEntryV2 united = GroupsValidationEntryV2.unite(list);
+        assertEquals(entry, united);
+    }
+
+    @Test
+    public void splitAndUniteNullGroupEntryTest() {
```

Comment:
Combine the tests into one ParametrizedTest. This would significantly decrease amount of code.

Reply:
what kind of parameters will we have?
list of arguments like

```
static Stream<Arguments> arguments = Stream.of(
  Arguments.of(LogicalPortsValidationEntryV2.builder()
                .asExpected(true)
                .missing(buildLogicalPortsInfo(1, 500))
                .excess(buildLogicalPortsInfo(1000, 600))
                .proper(buildLogicalPortsInfo(2000, 700))
                .misconfigured(buildMisconfiguredLogicalPortsInfo(3000, 800))
                .build(), 4, 3, 5), // object, first chunk, other chunks, chunk count
  Arguments.of(LogicalPortsValidationEntryV2.builder()
                .asExpected(true)
                .missing(buildLogicalPortsInfo(1, 500))
                .excess(buildLogicalPortsInfo(1000, 600))
                .proper(buildLogicalPortsInfo(2000, 700))
                .misconfigured(buildMisconfiguredLogicalPortsInfo(3000, 800))
                .build(), 4, 3, 5), // object, first chunk, other chunks, chunk count
 ****);
```

looks not very readable.


same list, but with factory methods?

```
 Arguments.of(getHugeObject, 4, 3, 5), // object, first chunk, other chunks, chunk count
 Arguments.of(getEmptyObject, 4, 3, 5), // object, first chunk, other chunks, chunk count
``` 

Better, but again, not very readable.  You need to jump to test, factory and parameters to understand what is going on.
getEmptyObject and other methods will be used only one time. Why we need to create a method for it? Usually we create methods when code will be called several times or to reduce other method size. But test method has 10 lines size

Reply:
Yes, something like option #2 is what I was talking about. It is a java problem that you can not describe test input with variable names, unlike, e.g., kotlin. But with complete code it would look better:
```
@ParametrizedTest(Stream.of(
Arguments.of(getHugeObject, 4, 3, 5),
Arguments.of(getObjectWhichWontSplitToEqualParts, 5, 6, 7),
...
Arguments.of(getCorruptedListsObject, 1, 0, null)))
void splitAndUniteTest(GroupsValidationEntryV2 entryToSplit, int firstChunk, int otherChunks, int expectedChunksAmount) {...
```
> You need to jump to test, factory and parameters to understand what is going on.

You won't if your naming would be clear. Test arguments with description are below the list of arguments, on the same screen, within amount of lines equal to number of your tests (i.e. if you have 6 tests, you'll have parameters description `GroupsValidationEntryV2 entryToSplit, int firstChunk, int otherChunks, int expectedChunksAmount` within max 6 lines from your 'unreadable' parameter).

> getEmptyObject and other methods will be used only one time

I agree, that during review I mistakenly assumed that you use almost each GroupsValidationEntryV2 several times. But it says about bad readability of your approach. How much time do you need to realize, why do you need a different tests for these objects:
```
TypicalSwitch.builder()
.setPorts(24)
.setIsls(4)
.setManufacturer("Cisco")
.setMaxDowntime(2400)
.setConsumedPower(500)
.build()
vs
TypicalSwitch.builder()
.setPorts(4)
.setIsls(2)
.setManufacturer("Cisco")
.setMaxDowntime(3400)
.setConsumedPower(10)
.build()
```
?
Now, how much time do you need to realize, why do you need a different tests for these objects:
```
Switches.WITH_SHORT_DOWNTIME
vs
Switches.WITH_LONG_DOWNTIME
```
?